lib/golem/tests/pendulum.h

```diff
@@ -0,0 +1,82 @@
+#include <math.h>
+
+struct Position2D {
+    float x{};
+    float y{};
+};
+
+namespace pendulum {
+const float GRAVITY = 9.81f;
+const float MASS = 1.f;
+const float LENGTH = 42.f;
+const float FRICTION = 0.1f;
+
+template <size_t frequency>
+struct System {
+    float angle;
+    float velocity;
+
+    System(float starting_angle, float starting_velocity)
+        : angle(starting_angle)
+        , velocity(starting_velocity)
+    {
+    }
+
+    float measure() const
+    {
+        return angle;
+    }
+    void apply(const float& torque)
+    {
+        const float inertia = MASS * LENGTH * LENGTH;
+        const float acceleration = torque / inertia - GRAVITY * std::sin(angle) / LENGTH - FRICTION * velocity;
+        velocity += acceleration / frequency;
+        angle += velocity / frequency;
+    }
+};
+
+struct Kinematics {
+    Position2D pos;
+
+    Kinematics()
+    {
+        update(0);
+    }
+
+    Position2D get() const
+    {
+        return pos;
+    }
+    void update(const float& angle)
+    {
+        pos.x = LENGTH * std::cos(angle);
```

Comment:
Please use `std::cosf` instead of `std::cos` since we only have single precision FPU.

Reply:
good catch

- [ ] use cosf